ingestion/src/metadata/ingestion/source/metabase.py

```diff
@@ -0,0 +1,234 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import json
+import logging
+import traceback
+import uuid
+from typing import Iterable
+from urllib.parse import quote
+
+import requests
+
+from metadata.generated.schema.api.lineage.addLineage import AddLineage
+from metadata.generated.schema.entity.data.dashboard import Dashboard as Model_Dashboard
+from metadata.generated.schema.entity.data.table import Table
+from metadata.generated.schema.entity.services.dashboardService import (
+    DashboardServiceType,
+)
+from metadata.generated.schema.type.entityLineage import EntitiesEdge
+from metadata.generated.schema.type.entityReference import EntityReference
+from metadata.ingestion.api.common import Entity, WorkflowContext
+from metadata.ingestion.api.source import Source, SourceStatus
+from metadata.ingestion.models.table_metadata import Chart, Dashboard
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.ingestion.ometa.openmetadata_rest import MetadataServerConfig
+from metadata.ingestion.source.sql_alchemy_helper import SQLSourceStatus
+from metadata.ingestion.source.sql_source import SQLConnectionConfig, SQLSourceStatus
+from metadata.utils.helpers import get_dashboard_service_or_create
+
+HEADERS = {"Content-Type": "application/json", "Accept": "*/*"}
+
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class MetabaseConnectionConfig(SQLConnectionConfig):
+    service_type = "MySQL"
+    session_id: str = None
+    scheme = ""
+    database_service_name: str = None
+
+    def get_connection_url(self):
+        pass
+
+
+class MetabaseSource(Source[Entity]):
+    config: MetabaseConnectionConfig
+    metadata_config: MetadataServerConfig
+    status: SQLSourceStatus
+
+    def __init__(
+        self,
+        config: MetabaseConnectionConfig,
+        metadata_config: MetadataServerConfig,
+        ctx: WorkflowContext,
+    ):
+        super().__init__(ctx)
+        self.config = config
+        self.metadata_config = metadata_config
+        self.status = SQLSourceStatus()
+        params = {}
+        params["username"] = self.config.username
+        params["password"] = self.config.password.get_secret_value()
+        resp = requests.post(
+            self.config.host_port + "/api/session/",
+            data=json.dumps(params),
+            headers=HEADERS,
+        )
+        session_id = resp.json()["id"]
```

Comment:
what happens if the resp or login fails this will fail with there is no json() instead of giving a clear error back to the user we might throw error that won't make sense
another use case could be if the username or password is wrong and login fails